packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1717,17 +1719,19 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   @override
   void didPopNext(Route<dynamic> nextRoute) {
     super.didPopNext(nextRoute);
-    changedInternalState();
+    changedInternalState(nextRoute is! ModalRoute || nextRoute is! DialogRoute);
```

Comment:
I think the right half of this expression is redundant because DialogRoute is a ModalRoute.

Reply:
I thought so too, but it breaks without the later half.